source/index.ts

```diff
@@ -180,8 +180,8 @@ export class Conf<T extends any = any> implements Iterable<[keyof T, T[keyof T]]
 	@param {key|object} - You can use [dot-notation](https://github.com/sindresorhus/dot-prop) in a key to access nested properties. Or a hashmap of items to set at once.
 	@param value - Must be JSON serializable. Trying to set the type `undefined`, `function`, or `symbol` will result in a `TypeError`.
     */
+	set<Key extends keyof T>(key: Key | string, value: T[Key] | unknown): void;
```

Comment:
This should be two different overloads. This now allows you to set using `Key`, but also the wrong value.